core/src/test/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbeTest.java

```diff
@@ -150,4 +151,31 @@ void shouldRecognizeInvalidGitHubUrl() throws Exception {
         assertThat(r1.status()).isEqualTo(ResultStatus.FAILURE);
         assertThat(r1.message()).isEqualTo("The plugin SCM link is invalid");
     }
+
+    @Test
+    void shouldReturnCorrectScmFolderPath() throws IOException {
+        final Plugin p1 = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final GitHub github = mock(GitHub.class);
+        final String repositoryName = "jenkinsci/test-repo";
+
+        when(p1.getScm()).thenReturn("https://github.com/" + repositoryName);
+        when(p1.getDetails()).thenReturn(Map.of(
+            UpdateCenterPluginPublicationProbe.KEY, ProbeResult.success(UpdateCenterPluginPublicationProbe.KEY, "")
+        ));
+        when(ctx.getScmRepository()).thenReturn(Path.of("src/test/resources/jenkinsci/test-repo/test-nested-dir-1"));
+        when(ctx.getGitHub()).thenReturn(github);
+        GHRepository repository = mock(GHRepository.class);
+        when(github.getRepository(repositoryName)).thenReturn(repository);
+
+        when(p1.getName()).thenReturn("test-repo");
+
+        final SCMLinkValidationProbe probe = getSpy();
+        final ProbeResult r1 = probe.apply(p1, ctx);
+
+        assertThat(ctx.getScmFolderPath()).isEqualTo("src/test/resources/jenkinsci/test-repo/test-nested-dir-1/test-nested-dir-2");
```

Comment:
Not the correct value. Because the repository is `src/test/resources/jenkinsci/test-repo/test-nested-dir-1`, the plugin folder path should be `test-nested-dir-2`. If the test fails because of this, it means you need to fix your code, not the other way around.

```suggestion
        assertThat(ctx.getScmFolderPath()).isEqualTo("test-nested-dir-2");
```